wireshark/README.md

```diff
@@ -1,31 +1,91 @@
+1. [System Requirements](#system-requirements)
+2. [Quick Start Guide](#quick-start-guide)
+3. [Project Components](#project-components)
+4. [Running Wireshark with OQS](#running-wireshark-with-oqs)
+    - [Explanation of Docker Options](#explanation-of-docker-options)
+5. [Testing Quantum-Safe Protocols](#testing-quantum-safe-protocols)
+6[Build Configuration and Updates](#build-configuration-and-updates)
 
-## Quick start
+## System Requirements
 
-1) Be sure to have [docker installed](https://docs.docker.com/install).
-2) Run `docker build -t openquantumsafe/wireshark .` to create an QSC-enabled (codepoint and OID aware) wireshark docker image.
+- **Docker**: Ensure [Docker](https://docs.docker.com/get-docker/) is installed and running on your system.
+- **X-Window System (for GUI Display)**:
+    - **Linux**: Run `xhost +si:localuser:$USER` to allow Docker to access the display.
+    - **Windows/macOS**: Install an X server such as [VcXsrv](https://sourceforge.net/projects/vcxsrv/) (Windows)
+      or [XQuartz](https://www.xquartz.org/) (macOS) and start it, ensuring to **disable access control** and **disable
+      native OpenGL**.
 
-## Usage
+## Quick Start Guide
 
-Information how to use the image is [available in the separate file USAGE.md](USAGE.md).
+```bash
+git clone https://github.com/open-quantum-safe/oqs-demos
+cd oqs-demos/wireshark
+docker build -t wireshark-oqs .
+docker run --rm -it --net=host -e DISPLAY=<your_host_ip>:<your_display_port> -v /tmp/.X11-unix:/tmp/.X11-unix wireshark-oqs
+```
 
-## Build options
+Replace `<your_host_ip>` with your IP address (e.g., `192.168.x.x`) and `<your_display_port>` with your display port,
+typically `:0`.
 
-The Dockerfile provided allows for customization of the image built:
+## Project Components
 
-### WIRESHARK_VERSION
+1. **Dockerfile**: Builds Wireshark with OpenSSL, liboqs, and OQS provider.
+2. **generate_qsc_header.py**: Processes `oqs-provider/oqs-template/generate.yml` with the `qsc_template.jinja2` to generate `qsc.h`,
+defining post-quantum KEMs and SIGs for Wireshark.
 
-This permits changing the wireshark code base to be used. 
+## Running Wireshark
 
-Tested default value is "3.4.9".
+You can run the Wireshark Docker container on Linux, Windows, or macOS using the following command:
 
-### QSC_SSL_FLAVOR
+  ```bash
+  docker run --rm -it --net=host -e DISPLAY=<your_host_ip>:<your_display_port> -v /tmp/.X11-unix:/tmp/.X11-unix wireshark-oqs
+  ```
+  Replace `<your_host_ip>` with your IP address (e.g., `192.168.x.x`) and `<your_display_port>` with your display port,
+  typically `:0`.
 
-Different quantum-safe TLS implementations have different names for the same algorithms. This option permits switching between them. Permitted values are "oqs" and "wolfssl".
+### Explanation of Docker Options
 
-Default is "oqs".
+- `--net=host`: Shares the host network with the container.
```

Comment:
This actually is USAGE.md. I'd recommend creating it separately such as for us to simply copy the contents of that file over to docker hub so normal docker hub users find all required documentation there: See help there: https://hub.docker.com/repository/docker/openquantumsafe/wireshark/general